gateway/src/main/java/eu/arrowhead/core/gateway/thread/ConsumerSideServerSocketThread.java

```diff
@@ -137,6 +161,16 @@ public void init(final MessageProducer sender) {
 	public boolean isInitialized() {
 		return initialized;
 	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public boolean isCommunicationStarted() {
+		return communicationStarted;
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	public ZonedDateTime getLastInteractionTime() {
+		return this.lastInteraction;
```

Comment:
you should update lastInteraction inside the onMessage() before 

relayClient.getBytesFromMessage()